File: src/main/kotlin/net/dontdrinkandroot/gitki/wicket/page/user/UserPage.java
package net.dontdrinkandroot.gitki.wicket.page.user;

import net.dontdrinkandroot.gitki.wicket.page.DecoratorPage;
import org.apache.wicket.model.IModel;
import org.apache.wicket.request.mapper.parameter.PageParameters;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public abstract class UserPage<T> extends DecoratorPage<T>
{
    public UserPage(PageParameters parameters) {
        super(parameters);
    }

    public UserPage(IModel<T> model) {
        super(model);
    }
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/wicket/page/user/UserPage.kt
package net.dontdrinkandroot.gitki.wicket.page.user

import net.dontdrinkandroot.gitki.wicket.page.DecoratorPage
import org.apache.wicket.model.IModel
import org.apache.wicket.request.mapper.parameter.PageParameters

abstract class UserPage<T> : DecoratorPage<T> {
    constructor(parameters: PageParameters) : super(parameters)
    constructor(model: IModel<T>) : super(model)
}